src/CAPI.cpp

```diff
@@ -0,0 +1,165 @@
+#include <algorithm>
+
+#include "CAPI.h"
+#include "ClientImpl.h"
+#include "TableIterator.h"
+#include "Row.hpp"
+
+struct c_client {
+   voltdb::ClientConfig* client_config;
+   voltdb::Client* client;
+};
+
+struct c_procedure {
+   voltdb::Procedure* procedure;
+   int numParams;
+   std::vector<voltdb::Parameter>* parameters;
+};
+
+struct c_invocation_response {
+   voltdb::InvocationResponse* response;
+};
+
+c_client* c_create_client(const char* usrname, const char* pwd,
+      const char* hostname, unsigned short port, bool keepConnecting,
+      /*voltdb::ClientAuthHashScheme schema,*/ bool enableAbandon, bool enableTimeout,
+      int timeoutInSec, bool useSSL) {
+   voltdb::ClientConfig* client_config = new voltdb::ClientConfig(
+         usrname, pwd, voltdb::HASH_SHA1/*schema*/, enableAbandon,
+         enableTimeout, timeoutInSec, useSSL);
+   voltdb::Client* client = new voltdb::Client(new voltdb::ClientImpl(*client_config));
```

Comment:
What happens if this or any other C++ call throws an exception a C program or anything using the C wrapper wouldn't handle the exception and I think it would be ignored.

Reply:
Good point, which I chose to ignore but shouldn't.
I'll use NULL with errno to address this.